6-Typescript/server/controllers/todoController.ts

```diff
@@ -1,78 +1,66 @@
-import { ITodo } from './../interfaces/ITodo';
-import {Request, Response} from 'express'
-import { ITodoService } from '../interfaces/ITodoService';
+import { ITodo } from "../interfaces/interfaces";
+import { Request, Response } from "express";
+import { ITodoService } from "../interfaces/interfaces";
 export class TodoController {
-    db: ITodoService
+    private db: ITodoService;
     constructor(db: ITodoService) {
-        this.db = db
-
+        this.db = db;
     }
     getTodoById = async (req: Request, res: Response): Promise<void> => {
         const userId: string = req.body.userId;
-        const todoId: string= req.params.id
-        
-        const todo: ITodo | null = await this.db.modelGetTodoById(userId, todoId);
-        if(todo) {
-            res.status(200).json(todo)
-        } else {                                                
-            res.sendStatus(404)
+        const todoId: string = req.params.id;
+
+        const todo: ITodo | null = await this.db.modelGetTodoById(
+            userId,
+            todoId
+        );
+        if (todo == null) {
+            res.sendStatus(404);
+            return;
         }
-    }
-    postTodo = async (req: Request, res: Response): Promise<void> => {   
+        res.status(200).json(todo);
+    };
+    postTodo = async (req: Request, res: Response): Promise<void> => {
+        const newTodo = req.body;
         try {
-            const {userId, todoText, todoId, checked}:
-            { userId:string, todoText:string, todoId:string, checked:boolean }
-             = req.body
-            const newTodo: ITodo = {
-                todoText: todoText,
-                todoId: todoId,
-                checked: checked,
-                userId: userId,
-            }
             const todo: ITodo | null = await this.db.modelPostTodo(newTodo);
-            if(!todo) {
-                res.status(400).send('Failed to post todo');
-            } else {
-                res.sendStatus(201);
+            res.sendStatus(201);
+        } catch (err: any) {
+            if (err.name == "ValidationError") {
```

Comment:
use an enum for this name. it can be changed later to `validationErrorIsNotFun` and you'll need to change it in few places (or forget, and then you'll have an error)

Reply:
sounds good!